### PR TITLE
Cover tmt.checks, tmt.frameworks and tmt.log with pyright

### PR DESCRIPTION
**Pull request by happz**
Pull Request Checklist

* [x] implement the feature

### CODE REVIEW

**Review by thrix — APPROVED** [review]
LGTM

File: tmt/checks/__init__.py [hunk 1]
```diff
@@ -251,9 +254,13 @@ def normalize_checks(
         return [normalize_test_check(key_address, raw_checks, logger)]
 
     if isinstance(raw_checks, list):
+        # ignore[redundant-cast]: mypy infers the type to be `List[Any]` while
+        # pyright, not makign assumptions about the type of items, settles for
```
Comment (janhavlin) [inline tmt/checks/__init__.py:258]:
```suggestion
        # pyright, not making assumptions about the type of items, settles for
```

File: pyproject.toml [hunk 2]
```diff
@@ -215,9 +215,9 @@ ignore = [
     "docs/**",
     "examples/**",
     "tests/**",
-    "tmt/checks/*.py",
+    # "tmt/checks/*.py",
```
Comment (psss) [inline pyproject.toml:218]: Why not just remove them from the `ignore` list?

**Review by psss — CHANGES_REQUESTED** [review]
Looks good, just one cleanup nitpick.

**Review by lukaszachy — APPROVED** [review]

**Comment by psss** [comment]
Could you please rebase on the latest `main`?

File: tmt/checks/__init__.py (hunk 1 above)
Comment (happz) [reply tmt/checks/__init__.py:246]: Fixed during rebase.

File: pyproject.toml (hunk 2 above)
Comment (happz) [reply pyproject.toml]: Fixed during rebase.

**Comment by happz** [comment]
@psss there's https://github.com/teemtee/tmt/pull/2455 standing between this PR and `main`. Rebased both of them...

**Comment by happz** [comment]
@psss and I rebased also https://github.com/teemtee/tmt/pull/2485. It would be nice if the whole sequence could be merged without no other PRs getting between them, it'd make rebase much easier.

**Review by psss — APPROVED** [review]

**Pull request merged**